origin/metrics.go

```diff
@@ -28,162 +30,162 @@ type muxerMetrics struct {
 }
 
 type TunnelMetrics struct {
-	haConnections     prometheus.Gauge
-	totalRequests     prometheus.Counter
-	requestsPerTunnel *prometheus.CounterVec
+	haConnections prometheus.Gauge
+
+	requests *prometheus.CounterVec
 	// concurrentRequestsLock is a mutex for concurrentRequests and maxConcurrentRequests
 	concurrentRequestsLock      sync.Mutex
 	concurrentRequestsPerTunnel *prometheus.GaugeVec
-	// concurrentRequests records count of concurrent requests for each tunnel
-	concurrentRequests             map[string]uint64
+	// concurrentRequests records count of concurrent requests for each tunnel, keyed by hash of label values
+	concurrentRequests             map[uint64]uint64
 	maxConcurrentRequestsPerTunnel *prometheus.GaugeVec
-	// concurrentRequests records max count of concurrent requests for each tunnel
-	maxConcurrentRequests map[string]uint64
+	// concurrentRequests records max count of concurrent requests for each tunnel, keyed by hash of label values
+	maxConcurrentRequests map[uint64]uint64
 	timerRetries          prometheus.Gauge
-	responseByCode        *prometheus.CounterVec
-	responseCodePerTunnel *prometheus.CounterVec
-	serverLocations       *prometheus.GaugeVec
+
+	responses       *prometheus.CounterVec
+	serverLocations *prometheus.GaugeVec
 	// locationLock is a mutex for oldServerLocations
 	locationLock sync.Mutex
 	// oldServerLocations stores the last server the tunnel was connected to
-	oldServerLocations map[string]string
+	oldServerLocations map[uint64]string
 
 	muxerMetrics *muxerMetrics
 }
 
-func newMuxerMetrics() *muxerMetrics {
+func newMuxerMetrics(baseMetricsLabels []string) *muxerMetrics {
 	rtt := prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Name: "rtt",
+			Name: "argotunnel_rtt",
 			Help: "Round-trip time in millisecond",
 		},
-		[]string{"connection_id"},
+		append(baseMetricsLabels, "connection_id"),
 	)
 	prometheus.MustRegister(rtt)
 
 	rttMin := prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Name: "rtt_min",
+			Name: "argotunnel_rtt_min",
 			Help: "Shortest round-trip time in millisecond",
 		},
```

Comment:
Instead of the above, I would:
```
rttMin := NewGaugeVec(
    "rtt_min", 
    "Shortest round-trip time in milliseconds", 
    append(baseMetricsLabels, "connection_id")
)

// ...

func NewGaugeVec(name, help string, metricsLabels []string) prometheus.GaugeVec {
    const ARGO_TUNNEL_NAMESPACE = "argo_tunnel"

    return prometheus.NewGaugeVec(
        prometheus.GaugeOpts{
            Name: name,
            Namespace: ARGO_TUNNEL_NAMESPACE,
            Help: help,
        },
        metricsLabels
    )
}
```

This helps keep the code DRY, and leaves less room for people to make mistakes.